### PR TITLE
uint: make max_value const, test, simplify

### PR DESCRIPTION
**Pull request by webmaster128**
~Based on #650 (merge this first)~

### CODE REVIEW

**Comment by ordian** [comment]
We should probably deprecate `max_value` instead as it's deprecated in std: https://doc.rust-lang.org/std/primitive.u64.html#method.max_value.

**Comment by webmaster128** [comment]
Right. Saw this too along the way. I deprecated it now.

File: uint/src/uint.rs [hunk 1]
```diff
@@ -800,13 +800,10 @@ macro_rules! construct_uint {
 			}
 
 			/// The maximum value which can be inhabited by this type.
+			#[deprecated(note = "replaced by the `MAX` associated constant on this type")]
```
Comment (ordian) [inline uint/src/uint.rs:803]:
```suggestion
			#[deprecated(since = "0.10.0", note = "replaced by the `MAX` associated constant on this type")]
```

**Review by ordian — APPROVED** [review]

File: uint/src/uint.rs (hunk 1 above)
Comment (webmaster128) [reply uint/src/uint.rs]: Sure, committed

File: uint/src/uint.rs [hunk 2]
```diff
@@ -800,13 +800,10 @@ macro_rules! construct_uint {
 			}
 
 			/// The maximum value which can be inhabited by this type.
+			#[deprecated(since = "0.10.0", note = "replaced by the `MAX` associated constant on this type")]
```
Comment (ordian) [inline uint/src/uint.rs:803]: Sorry about this, but the std lib doesn't deprecate it as of yet, but will in the future it seems.doc
```suggestion
```

File: uint/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog].
 
 ## [Unreleased]
 - Make `one` const. [#650](https://github.com/paritytech/parity-common/pull/650)
+- Make `max_value` const and deprecate it. [#652](https://github.com/paritytech/parity-common/pull/652)
```
Comment (ordian) [inline uint/CHANGELOG.md:9]:
```suggestion
- Make `max_value` const. [#652](https://github.com/paritytech/parity-common/pull/652)
```

**Review by ordian — APPROVED** [review]

File: uint/src/uint.rs (hunk 2 above)
Comment (webmaster128) [reply uint/src/uint.rs:803]: Oh, they have a _will be deprecated_ state. Strange.

**Pull request merged**